backend/hitas/views/thirty_year_regulation.py

```diff
@@ -1,18 +1,56 @@
+from uuid import UUID
+
+from django.http import HttpResponse
 from rest_framework import status
+from rest_framework.decorators import action
 from rest_framework.exceptions import ValidationError
+from rest_framework.request import Request
 from rest_framework.response import Response
 from rest_framework.viewsets import ViewSet
 
-from hitas.services.thirty_year_regulation import perform_thirty_year_regulation
+from hitas.exceptions import ModelConflict
+from hitas.models.thirty_year_regulation import RegulationResult
+from hitas.services.thirty_year_regulation import (
+    get_thirty_year_regulation_results_for_housing_company,
+    perform_thirty_year_regulation,
+)
 from hitas.utils import from_iso_format_or_today_if_none
+from hitas.views.utils.pdf import get_pdf_response
 
 
 class ThirtyYearRegulationView(ViewSet):
-    def list(self, request, *args, **kwargs) -> Response:
+    def list(self, request: Request, *args, **kwargs) -> Response:
         try:
-            calculation_date = from_iso_format_or_today_if_none(kwargs.get("calculation_date"))
+            calculation_date = from_iso_format_or_today_if_none(request.query_params.get("calculation_date"))
         except ValueError as error:
             raise ValidationError({"calculation_date": str(error)}) from error
 
         results = perform_thirty_year_regulation(calculation_date)
         return Response(data=results, status=status.HTTP_200_OK)
+
+    @action(
+        methods=["GET"],
+        detail=False,
+        url_path=r"reports/regulation-letter",
+        url_name="letter",
```

Comment:
`url_name` concatenates with the base URL reverse name(?) to give the actual reverse name. So in this case it's `thirty-year-regulation-letter` if just `letter` is used as `url_name`, otherwise, it would be `thirty-year-regulation-regulation-letter`. I would keep it short.

I'll change the URL to match.